hwloc/topology-rsmi.c

```diff
@@ -0,0 +1,268 @@
+/*
+ * Copyright © 2012-2020 Inria.  All rights reserved.
+ * Copyright (c) 2020, Advanced Micro Devices, Inc. All rights reserved.
+ * Written by Advanced Micro Devices,
+ * See COPYING in top-level directory.
+ */
+
+#include "private/autogen/config.h"
+#include "hwloc.h"
+#include "hwloc/plugins.h"
+
+/* private headers allowed for convenience because this plugin is built within hwloc */
+#include "private/misc.h"
+#include "private/debug.h"
+
+#include <dlfcn.h>
+#include <rocm_smi.h>
+
+/**
+ * PCI information about a GPU device.
+ */
+typedef struct rsmiPciInfo_st {
+  union {
+    struct {
+#ifdef HWLOC_BIGENDIAN
+      uint64_t domain : 32;
+      uint64_t reserved : 16;
+      uint64_t bus : 8;
+      uint64_t device : 5;
+      uint64_t function : 3;
+#else
+      uint64_t function : 3;
+      uint64_t device : 5;
+      uint64_t bus : 8;
+      uint64_t reserved : 16;
+      uint64_t domain : 32;
+#endif
+    };
+    uint64_t bdfid;
+  };
+} rsmiPciInfo_t;
```

Comment:
Looks like we need to add some configury to define HWLOC_BIGENDIAN?
Also isn't this structure defined in RSMI headers? I'd be better not to duplicate it in hwloc.

Reply:
This structure isn't defined in RSMI headers. We can remove HWLOC_BIGENDIAN, since it's not used.

Reply:
Oh I see, only the bdfid part of union seems to be passed to rsmi_dev_pci_id_get(), the struct is here for decoding only. Does RSMI work on some big endian architectures?

Reply:
Yes, that's for decoding only. I have ashed some regarding big endian architectures.

Reply:
Given that the rsmi doc explicitly says:
```
 *      BDFID = ((DOMAIN & 0xffffffff) << 32) | ((BUS & 0xff) << 8) |
 *                                   ((DEVICE & 0x1f) <<3 ) | (FUNCTION & 0x7)
```
I would just remove the structure entirely and decode the uint64_t manually, it won't depend on big/little endian.

Reply:
Ok